### ISSUE TITLE
Rework Locations side bar

### ISSUE THREAD

**Issue by nikitabobko** [sublimehq/sublime_merge#479]
**Problem description**

* I don't like that double click on branch name in side bar do `git checkout` as it changes the state and can be potentially dangerous.
* eye icon is toooo small. Too hard to hit it
* Alt + eye icon shows only current branch and it makes sense as it's very frequent use case to show only current branch. But single eye icon click hides current branch which is useless. So more frequent action requires you to do Alt + click but useless operations is more easy to do just click on this eye icon. So it's also problem

**Preferred solution**

* Allow to setup action which performs on branch single/double click (different people have different cases some do checkout frequently than showing only particular branch and vise versa)
* Rework this eye icon (Make it bigger or remove it at all). This eye icon is controversial as I described that more frequently action = Alt + click, but useless action is just single click.

**Comment by deathaxe** [comment]
> I don't like that double click on branch name in side bar do `git checkout`

I often use the sidebar to exactly do that as I prefere that over struggling with the always different looking complicated to read context menues in the commit tree, which are barely usable.

I never liked them and don't get used to them.

**Comment by nikitabobko** [comment]
@deathaxe It's ok that people have different use cases. You prefer this behavior. Someone else may prefer it too. But I'd prefer another behavior. I don't say that `git checkout` on double click is wrong behavior, I just say that it'd be better to have this option configurable.

**Comment by nikitabobko** [comment]
What's under question is which behavior should be default. I have concern that it's potentially   dangerous as it changes state and when you firstly use Sublime Merge it could be not obvious that it will do `git checkout` on double click. But if author assumes that `git checkout` is used more often than showing only current branch then I won't argue. Having configurable option is already fine.

**Comment by deathaxe** [comment]
> It's ok that people have different use cases

This is just what I wanted to express.

> that current behavior is potentially dangerous

`git checkout` works only if is safe. If your working directory contains uncommitted changes, which would get lost git rejects the checkout request anyway.

> But if Sublime Merge's author assumes that git checkout is used more often than showing only current branch then I won't argue

The current implementation provides both options.

1. Single click -> view branch
2. Double click -> checkout branch

**Comment by nikitabobko** [comment]
> The current implementation provides both options.

No. I mean not jumping to particular commit but showing commits only from selected branch (the way how now `Alt + click` works). I'd like to see option to double click on branch to show commits only from selected branch. And one more double click to return back and show all commits.
